zulipterminal/ui_tools/boxes.py

```diff
@@ -893,14 +907,20 @@ def keypress(self, size: urwid_Size, key: str) -> Optional[str]:
                     # We extract recipients' user_ids and emails only once we know
                     # that all the recipients are valid, to avoid including any
                     # invalid ones.
-                    self.update_recipients(self.to_write_box)
+                    self.update_recipients_from_emails_in_widget(self.to_write_box)
 
             if not self.msg_body_edit_enabled:
                 return key
             if self.focus_position == self.FOCUS_CONTAINER_HEADER:
                 self.focus_position = self.FOCUS_CONTAINER_MESSAGE
             else:
-                self.focus_position = self.FOCUS_CONTAINER_HEADER
+                if self.compose_box_status == "open_with_stream":
```

Comment:
This logic is a bit tricky to read. I was thinking about refactoring the whole `CYCLE_COMPOSE_FOCUS` section, but that might be overkill for this PR.